src/xdp/trn_transit_xdp.c

```diff
@@ -460,8 +460,50 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 				return XDP_ABORTED;
 			}
 		} else {
+			pkt->pod_label_value_opt = (void *)pkt->scaled_ep_opt + sizeof(*pkt->scaled_ep_opt);
+	
+			if (pkt->pod_label_value_opt + 1 > pkt->data_end) {
+				bpf_debug("[Scaled_EP:%d:0x%x] ABORTED: Bad offset\n", __LINE__,
+					bpf_ntohl(pkt->itf_ipv4));
+				return XDP_ABORTED;
+			}
+
+			if (pkt->pod_label_value_opt->opt_class != TRN_GNV_OPT_CLASS) {
+				bpf_debug("[Scaled_EP:%d:0x%x] ABORTED: Unsupported Geneve option class\n",
+					__LINE__, bpf_ntohl(pkt->itf_ipv4));
+				return XDP_ABORTED;
+			}
+
+			if (pkt->pod_label_value_opt->type != TRN_GNV_LABEL_VALUE_OPT_TYPE) {
+				bpf_debug("[Scaled_EP:%d:0x%x] ABORTED: Unsupported Geneve option type\n",
+					__LINE__, bpf_ntohl(pkt->itf_ipv4));
+				return XDP_ABORTED;
+			}
+
+			pkt->namespace_label_value_opt = (void *)pkt->pod_label_value_opt + sizeof(*pkt->pod_label_value_opt);
+
+			if (pkt->namespace_label_value_opt + 1 > pkt->data_end) {
+				bpf_debug("[Scaled_EP:%d:0x%x] ABORTED: Bad offset\n", __LINE__,
+					bpf_ntohl(pkt->itf_ipv4));
+				return XDP_ABORTED;
+			}
+
+			if (pkt->namespace_label_value_opt->opt_class != TRN_GNV_OPT_CLASS) {
+				bpf_debug("[Scaled_EP:%d:0x%x] ABORTED: Unsupported Geneve option class\n",
+					__LINE__, bpf_ntohl(pkt->itf_ipv4));
+				return XDP_ABORTED;
+			}
+
+			if (pkt->namespace_label_value_opt->type != TRN_GNV_LABEL_VALUE_OPT_TYPE) {
```

Comment:
I suggest changing this to two separate types: TRN_GNV_POD_LABEL_VALUE_OPT_TYPE & TRN_GNV_NAMESPACE_LABEL_VALUE_OPT_TYPE. This will clearly distinguish them when we capture them on the wire. And will help in the future when we figure out how to add only the types that we want to send instead of always bloating up the packet whether or not we have labels.

Reply:
Splitted as suggested